db/schemas.py

```diff
@@ -71,10 +75,65 @@ def get_all_schemas(engine):
     return inspector.get_schema_names()
 
 
+def get_related_tables(engine, schema):
+    metadata = MetaData()
+    pg_tables = Table("pg_tables", metadata, autoload_with=engine)
+    sel = (
+        select(pg_tables.c.schemaname, pg_tables.c.tablename).
+        where(pg_tables.c.schemaname == schema)
+    )
+    with engine.begin() as conn:
+        results = conn.execute(sel).fetchall()
+
+    related_tables = [tables.reflect_table(table, schema, engine)
+                      for schema, table in results]
+    return related_tables
+
+
 def create_schema(schema, engine):
     """
     This method creates a Postgres schema.
     """
     if schema not in get_all_schemas(engine):
         with engine.begin() as connection:
             connection.execute(CreateSchema(schema))
+
+
+def delete_schema(schema, engine, cascade=False):
+    """
+    This method deletes a Postgres schema.
+    """
+    related_tables = get_related_tables(engine, schema)
+    if schema in get_all_schemas(engine):
+        with engine.begin() as connection:
+            try:
+                connection.execute(DropSchema(schema, cascade=cascade))
+            except InternalError as e:
+                if isinstance(e.orig, DependentObjectsStillExist):
+                    return related_tables
+                else:
+                    raise e
+    return related_tables
+
+
+class RenameSchema(DDLElement):
+    def __init__(self, schema, rename_to):
+        self.schema = schema
+        self.rename_to = rename_to
+
+
+@compiler.compiles(RenameSchema)
+def compile(element, compiler, **_):
+    return "ALTER SCHEMA %s RENAME TO %s" % (
+        element.schema,
+        element.rename_to
+    )
+
+
+def rename_schema(schema, engine, rename_to):
+    """
+    This method deletes a Postgres schema.
+    """
+    if schema in get_all_schemas(engine):
+        with engine.begin() as connection:
```

Comment:
Have you verified that this updates references to the schema from other schemas? (e.g. if there's a foreign key to a table in this schema from a different schema)

Reply:
Please add a test to verify behavior.